docs/guides/debugging.md

```diff
@@ -0,0 +1,220 @@
+
+When scraping live pages directly, especially when comparing with results produced earlier, be wary that the live page may have changed between both scrapings. So differences in results may also come from differences in the actual content that is checked.
+
+## Reproducing results
+
+Once you have a page, you can load it into Alfa and re-run the audit. The usual case is to investigate the result of a given rule, which can for example be done with the following script:
+
+```typescript
+import { Outcome } from "@siteimprove/alfa-act";
+import { Array } from "@siteimprove/alfa-array";
+import { Rules } from "@siteimprove/alfa-rules";
+import { Page } from "@siteimprove/alfa-web";
+import * as fs from "fs";
+
+// Select rule and page
+const rule = "R68";
+const file = "./my-page.json";
+
+// De-serialize page
+const page = Page.from(JSON.parse(fs.readFileSync(file, "utf-8")));
+
+main();
+
+async function main() {
+  // Run the rule and sort results.
+  const result = Array.from(await Rules.get(rule).getUnsafe().evaluate(page));
+  const cantTell = result.filter(Outcome.isCantTell);
+  const passed = result.filter(Outcome.isPassed);
+  const failed = result.filter(Outcome.isFailed);
+
+  // Output some basic information
+  console.log(
+    `cantTell: ${cantTell.length}, passed: ${passed.length}, failed: ${failed.length}`
+  );
+
+  // Uncomment next line to list all failed results. Warning! This may be very long…
+  // console.dir(failed);
+}
+```
+
+Save this in the `scratches` directory, for example as `page.ts` and then, from the top-level Alfa directory, run:
+
+```shell
+$ yarn build
+$ node scratches/page.js
+```
+
+(the file is saved as `.ts`, but, after building, it is the `.js` that must be run)
+
+The rule to check, and the file to load, are configured after the `import` block. The Alfa page is de-serialized from the JSON file. Then a simple audit of a given rule is run and simple statistics are shown, which allow to confirm that the numbers are the same…
+
+## Time travel
+
+Since we run into the Github repo, it is very easy to test with an older version of Alfa. As long as only the `scratches` directory has been changed, just `git checkout vX.Y.Z` will go back to any version (`yarn install` and `yarn build` need to be rerun). This makes it easy to check when a breaking change has been introduced. Be wary that by going back too far in time, the `scratches/page.ts` script itself might break if it's using more modern constructions…
+
+In case of regression, or change of behavior, finding the precise version where it was introduced and then searching for cues in the [Changelog](../../CHANGELOG.md) is often a good way to narrow dowrn the possible cause.
```

Comment:
Made me think of [`git bisect`](https://git-scm.com/docs/git-bisect) which does a binary search of the history to find the commit where e.g. a bug was introduced. It can be used interactively where the user manually checks if the bug is present and it can even be automated which sounds like something we could do with the scratch file as long as it doesn't break. Just a thought, doesn't need to be included in the guide 🙂